### DISCUSSION TITLE
关于Vue中作用域插槽的使用方法

### DISCUSSION THREAD

**Discussion by brandonhulala** [alibaba/formily#2514] [Q&A 2.x]
我先看了一遍官方文档，里面关于作用域插槽的demo，都是基于functional为true的组件，没有基于defindComponent出来的组件。

然后，我试了一下，发现在使用defindComponent方法定义组件的时候，调用h函数时传入的第二个参数是一个json对象，可以给这个json设置attrs和on，但是再设置一个scopedSlots，就不生效。但是，在functional组件中， 这个scopedSlots可以生效。

所以，我的问题来了，是不是只要涉及到作用域插槽，就不能使用defindComponent组件，只能使用functional组件，是不是这样呀？

**Comment by muuyao** [comment]
不是这样的，可以给个demo么？

**Reply by brandonhulala** [reply]
我想先问一下，为啥官方文档中关于“vue作用域插槽”的demo，都是用functional组件写的，为啥不用defindComponent组件

**Reply by muuyao** [reply]
只是vue2 里用 functional 性能好一些，而且官方文档里例子都比较简单，组件没有自己的状态
